app/controllers/tags_controller.rb

```diff
@@ -1,22 +1,16 @@
 class TagsController < ApplicationController
-
   def index
-    @tags =  Tag.search(params[:q])
-  end
-
-  def popular
-    render json: Tag.get_tags_that_are(:popular), status: 200
-  end
-
-  def recent
-    render json: Tag.get_tags_that_are(:recent), status: 200
+    render partial: 'tag_properties', locals: { tags: Tag.search(params[:q])}
```

Comment:
Could you shed more light on the **low-level** nature of the use of `locals`? The [render documentation](http://apidock.com/rails/ActionController/Base/render) is not clear it's possibly non-conventional use.

Using an instance variable would be a great idea. But it's not as if the information is used in multiple places, just one. Do you think it's worth littering the namespace with instance variable(s) for lack of a better way to pass information from controller to view?